### PR TITLE
feat(unlock-app) - Footer component

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
New component footer that will be used across all out project.

## How the component works
Screenshot 2022-12-07 at 17 28 43

## Component without Newsletter part

- The component has a transparent background in order to adapt to different backgrounds 
Screenshot 2022-12-07 at 18 53 55

Screenshot 2022-12-07 at 18 26 57

## Hover state

Screenshot 2022-12-08 at 16 34 02

Screenshot 2022-12-08 at 16 36 49

Screenshot 2022-12-08 at 16 37 32

# Issues

Fixes https://github.com/unlock-protocol/unlock/issues/10500
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react'
+import { Button } from '@unlock-protocol/ui'
+
+export const EMAIL_SUBSCRIPTION_FORM = {
+  portalId: '19942922',
+  formGuid: '868101be-ae3e-422e-bc86-356c96939187',
+}
```
Comment (searchableguy) [inline packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx:7]: Should not be in UI library. This should be passed as props.

File: packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      )
+      return
+    }
+    setConfirm(true)
+    // Delay before reseting the form for user to notice the impact
+    setTimeout(() => {
+      setConfirm(false)
+      setEmail('')
+    }, 3000)
+  }
```
Comment (searchableguy) [inline packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx:52]: Hm, I would not put it here but abstract it out to a function you can pass. This makes the component usable and easy to change.

File: packages/ui/lib/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+const Footer = ({ privacyUrl, termsUrl, menuItems }: FooterProps) => {
+  return (
+    <footer className="flex flex-col gap-24">
+      <NewsletterBox />
+      <div className="flex flex-col gap-16 mb-20 md:grid md:grid-cols-3 md:gap-44">
+        <div className="flex flex-col gap-10">
+          <img
+            className="self-start h-10"
+            src="/images/unlock-footer-logo.svg"
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply packages/ui/lib/components/Footer/Footer.tsx:93]: ah! you are right!

[the remark this answers is not in the document]
Comment (searchableguy) [reply packages/ui/lib/components/Footer/Footer.tsx]: Sorry, by importing as component - I mean 

either copy the contents of the svg into a component

```jsx

export const Icon = () => {
  return <svg> ... </svg>
}
```

or add a loader to vite so you can 
```jsx
import Icon from "../svg/icon.svg"

<Icon />
```

File: packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx (hunk 1 above)
Comment (searchableguy) [reply packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx:7]: Yes but these are re-usable components. The package is published to npm. We should not add these constants which we use directly.

Comment (julien51) [reply packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx:7]: I agree with @searchableguy here.

File: packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        fields: [
+          {
+            name: 'email',
+            value: email,
+          },
+        ],
+      }),
+    }
+
+    const endpoint = `https://api.hsforms.com/submissions/v3/integration/submit/${portalId}/${formGuid}`
```
Comment (julien51) [inline packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx:37]: We could make the whole endpoint by just a prop... So it does not require the use of Hubspot.

File: packages/ui/lib/components/Footer/Footer.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          fields: [
+            {
+              name: 'email',
+              value: email,
+            },
+          ],
+        }),
+      }
+
+      await fetch(endpoint, options)
```
Comment (searchableguy) [inline packages/ui/lib/components/Footer/Footer.stories.tsx:35]: don't need this implementation in storybook. will end up causing false submissions.

File: packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
...fill='none' preserveAspectRatio='xMinYMin' className='inset-0 max-w-full max-h-full' %3E%3Cg clip-path='url(%23clip0_8422_11409)'%3E%3Crect width='1200' height='210' fill='%23BFB1F7' /%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M-47.9538 243.873C-46.4252 239.306 -43.685 233.435 -43.1176 232.03C-41.4088 227.789 -39.8302 223.494 -37.9892 219.308C-35.899 214.555 -33.5264 209.929 -31.3299 205.223C-23.9901 189.483 -15.1091 175.018 -7.53086 159.559C12.9188 117.844 56.9639 57.9278 99.901 35.279C128.878 19.9944 181.985 37.919 190.813 71.071C196.254 91.5061 179.553 109.633 165.213 122.414C160.114 126.959 155.016 131.502 149.917 136.047C148.898 136.956 146.858 138.774 146.858 138.774C146.858 138.774 156.084 130.551 160.701 126.435C168.04 119.894 175.598 114.182 184.588 110.891C206.479 102.874 236.004 123.028 242.631 143.066C253.85 176.993 231.551 201.843 207.268 223.487C160.779 264.922 87.0659 295.761 25.3885 302.733C-6.10637 306.293 -30.019 300.618 -50.9946 277.084C-57.3145 269.993 -49.8737 249.606 -47.9538 243.873Z' fill='%23D8F177' /%3E%3C/g%3E%3Cdefs%3E%3CclipPath id='clip0_8422_11409'%3E%3Crect width='1200' height='210' fill='white' /%3E%3C/clipPath%3E%3C/defs%3E%3C/svg%3E")`,
```
Comment (julien51) [inline packages/ui/lib/components/Footer/EmailSubscriptionForm.tsx:5]: Let's just put this in `.svg` file :)

**Review by julien51 — APPROVED** [review]

**Review by julien51 — APPROVED** [review]
LGTM

File: packages/ui/lib/components/Link/Link.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+      </NextLink>
+    )
+  }
+
+  if (isAnchorLink) {
+    return <a href={href} {...rest} />
+  }
+
+  return <a target="_blank" rel="noopener noreferrer" href={href} {...rest} />
+}
```
Comment (searchableguy) [inline packages/ui/lib/components/Link/Link.tsx:21]: Add nextjs as peer dependency.

File: packages/ui/package.json [hunk 8]
```diff
@@ -44,6 +44,7 @@
     "tailwindcss": "3.2.4"
   },
   "peerDependencies": {
+    "next": "12.3.3",
```
Comment (searchableguy) [inline packages/ui/package.json:47]:
```suggestion
    "next": "12.x.x",
```

**Review by searchableguy — APPROVED** [review]

**Pull request merged**